core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java

```diff
@@ -0,0 +1,101 @@
+/*
```

Comment:
I think this deserves a separate PR, WDYT ? We need to document how this works and make sure that we add all the warnings regarding the restrictions of using this filter in conjunction with others.